ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-reference-addon.yaml

```diff
@@ -0,0 +1,44 @@
+base_images:
+  cli:
+    name: "4.13"
+    namespace: ocp
+    tag: cli
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.17
+images:
+- dockerfile_literal: |
+    FROM quay.io/redhat_msi/ocp-addons-operators-cli:latest
```

Comment:
What code base is being used to create this image?

Reply:
This repo holds the code base: https://github.com/RedHatQE/ocp-addons-operators-cli

Reply:
I wasn't able to see any CI system that is pushing this code to quay is that available somewhere that I just missed?

Reply:
We have a `Build Trigger` in quay which builds a new `latest` image tag on push to the default branch:
https://quay.io/repository/redhat_msi/ocp-addons-operators-cli?tab=builds

Reply:
Ah ok thanks for sharing. So if I understand this right 
1. The code will merge in the github repo
2. This triggers an image build tagged as latest in Quay
3. Then with this config we are telling the OCP build cluster to build this image again and push it to the build clusters registry (ie. `registry.build01.ci.openshift.org/ci-op-6q2bqinp/pipeline`).
4. Which is then used as the container env for the ref.

A couple of questions:
1. What happens when the Quay build fails due to an issue in the latest merged code from https://github.com/RedHatQE/ocp-addons-operators-cli?
2. Instead of building the image again each time that this job runs, can we push the image to the OCP CI registries ahead of time so that the image we want to work with already exists? If we like this functionality we can do this image push ahead of time two different ways.
      1. We can mirror the quay image to OCP ci's registry like @chetna14manku is doing [here](https://github.com/openshift/release/pull/37565/files#diff-03fc359cab8a8f80758a860fb8a32b1d5013b93b04ab910fc0f17ebf41ac12a0)
      2. Or we can remove Quay from this workflow and setup the repo https://github.com/RedHatQE/ocp-addons-operators-cli to test the image build when a PR is submitted to verify that the change isn't breaking the image. Then on merge it will push the image to the OCP CI registries to be used as a base_image in this config. (example of this is [here](https://cspi-qe.gitbook.io/ocp-ci-onboarding/onboarding/onboarding_guide/scenario_development_guide#so-how-do-we-solve-this) with links, this isn't specific to private repos though working with private repos is what uncovered this solution).

Reply:
3 and 4 are not relevant anymore, check https://github.com/openshift/release/pull/37752